src/main/java/org/elasticsearch/script/javascript/JavaScriptScriptEngineService.java

```diff
@@ -41,258 +45,287 @@
-        }
-
-        @Override
-        public Object unwrap(Object value) {
-            return ScriptValueConverter.unwrapValue(value);
-        }
-    }
-
-    public static class JavaScriptSearchScript implements SearchScript {
-
-        private final Script script;
-
-        private final Scriptable scope;
-
-        private final SearchLookup lookup;
-
-        public JavaScriptSearchScript(Script script, Scriptable scope, SearchLookup lookup) {
-            this.script = script;
-            this.scope = scope;
-            this.lookup = lookup;
-        }
-
-        @Override
-        public void setScorer(Scorer scorer) {
-            lookup.setScorer(scorer);
-        }
-
-        @Override
-        public void setNextReader(AtomicReaderContext context) {
-            lookup.setNextReader(context);
-        }
-
-        @Override
-        public void setNextDocId(int doc) {
-            lookup.setNextDocId(doc);
-        }
-
-        @Override
-        public void setNextScore(float score) {
-            ScriptableObject.putProperty(scope, "_score", score);
-        }
-
-        @Override
-        public void setNextVar(String name, Object value) {
-            ScriptableObject.putProperty(scope, name, value);
-        }
-
-        @Override
-        public void setNextSource(Map<String, Object> source) {
-            lookup.source().setNextSource(source);
-        }
-
-        @Override
-        public Object run() {
-            Context ctx = Context.enter();
-            try {
-                ctx.setWrapFactory(wrapFactory);
-                return ScriptValueConverter.unwrapValue(script.exec(ctx, scope));
-            } finally {
-                Context.exit();
-            }
-        }
-
-        @Override
-        public float runAsFloat() {
-            return ((Number) run()).floatValue();
-        }
-
-        @Override
-        public long runAsLong() {
-            return ((Number) run()).longValue();
-        }
-
-        @Override
-        public double runAsDouble() {
-            return ((Number) run()).doubleValue();
-        }
-
-        @Override
-        public Object unwrap(Object value) {
-            return ScriptValueConverter.unwrapValue(value);
-        }
-    }
-
-    /**
-     * Wrap Factory for Rhino Script Engine
-     */
-    public static class CustomWrapFactory extends WrapFactory {
-
-        public CustomWrapFactory() {
+public class JavaScriptScriptEngineService extends AbstractComponent implements
+		ScriptEngineService {
+
+	private final AtomicLong counter = new AtomicLong();
+
+	private static WrapFactory wrapFactory = new CustomWrapFactory();
+
+	private final int optimizationLevel;
+
+	private Global globalScope;
+
+	private File modulesFile;
+
+	@Inject
+	public JavaScriptScriptEngineService(Settings settings, Environment env) {
+		super(settings);
+
+		this.optimizationLevel = componentSettings.getAsInt("optimization_level", 1);
```

Comment:
Looks like the different indent space introduced lots of differences. Could they be cleaned to see the main changes? Just my 2 cents. :smile: